### PR TITLE
Update types to reflect the names in proto

### PR DESCRIPTION
**Pull request by bogdandrutu**
NON_MONOTONIC_SUM will be soon merged with SUM and offer the "isMonotonic" as a property.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Review by breedx-splk — COMMENTED** [review]
The execution of the rename seems fine, and the comments were updated.  

Is the plan then to have `isMonotonic()` on all of the `Type`s, so that one can also ask `SUMMARY.isMonotonic()` or `GAUGE_LONG.isMonotonic()`?  That doesn't seem great...but maybe you had a different approach in mind?

File: exporters/logging/src/test/java/io/opentelemetry/exporter/logging/LoggingMetricExporterTest.java [hunk 1]
```diff
@@ -74,7 +74,7 @@ void testExport() {
                 "counterOne",
                 "A simple counter",
                 "one",
-                MetricData.Type.MONOTONIC_LONG,
+                MetricData.Type.SUM_LONG,
```
Comment (Oberon00) [inline exporters/logging/src/test/java/io/opentelemetry/exporter/logging/LoggingMetricExporterTest.java:77]: LONG_SUM would sound more natural IMHO.

**Comment by bogdandrutu** [comment]
@breedx-splk I will design in a same way as the proto, based on the type we return an Object that has the Points + extra metadata specific to that type (see https://github.com/open-telemetry/opentelemetry-proto/blob/master/opentelemetry/proto/metrics/v1/metrics.proto)

File: exporters/logging/src/test/java/io/opentelemetry/exporter/logging/LoggingMetricExporterTest.java (hunk 1 above)
Comment (bogdandrutu) [reply exporters/logging/src/test/java/io/opentelemetry/exporter/logging/LoggingMetricExporterTest.java:77]: good point. will change.

**Review by jkwatson — APPROVED** [review]
Thanks!

**Pull request merged**